N1C WAF Security Dashboard Release (#1834)
nginx/documentation: content/nginx-one-console/waf-integration/waf-security-dashboard/set-up-security-monitoring.md

---
nd-content-type: how-to
nd-docs: DOCS-000
nd-product: NONECO
title: Set up security monitoring
description: "Forward F5 WAF for NGINX security events to the NGINX One Console security monitoring dashboard."
weight: 300
toc: true
nd-keywords: "F5 WAF for NGINX, security monitoring, security dashboard, default log profile, security events"
nd-summary: >
  Forward F5 WAF for NGINX security events from an NGINX Plus instance to the NGINX One Console security monitoring dashboard.
  You deploy the `secops_dashboard` log profile through the console, add the WAF directives to your NGINX configuration, and verify events flow into the dashboard.
  Repeat these steps for each data plane you want to monitor.
nd-audience: operator
---

## Overview

> This guide walks you through enabling F5 WAF for NGINX security monitoring on an NGINX Plus instance that is already connected to NGINX One Console. After you complete the steps, security events appear in the [security monitoring dashboard]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/_index.md" >}}), where you can review attacks, violations, and triggered signatures.

Reviewer: is this plus only?

Reviewer 2: yeah. NGINX Plus constraint is now stated up front on the section landing page. Hope this helps and is a good place to call out.

---
nd-content-type: how-to
nd-docs: DOCS-000
nd-product: NONECO
title: Set up security monitoring
description: "Forward F5 WAF for NGINX security events to the NGINX One Console security monitoring dashboard."
weight: 300
toc: true
nd-keywords: "F5 WAF for NGINX, security monitoring, security dashboard, default log profile, security events"
nd-summary: >
  Forward F5 WAF for NGINX security events from an NGINX Plus instance to the NGINX One Console security monitoring dashboard.
  You deploy the `secops_dashboard` log profile through the console, add the WAF directives to your NGINX configuration, and verify events flow into the dashboard.
  Repeat these steps for each data plane you want to monitor.
nd-audience: operator
---

## Overview

> This guide walks you through enabling F5 WAF for NGINX security monitoring on an NGINX Plus instance that is already connected to NGINX One Console. After you complete the steps, security events appear in the [security monitoring dashboard]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/_index.md" >}}), where you can review attacks, violations, and triggered signatures.

Reviewer 3: ```suggestion
Use this guide to enable F5 WAF for NGINX security monitoring on an NGINX Plus instance that is already connected to NGINX One Console. After completing the steps, security events appear in the [security monitoring dashboard]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/_index.md" >}}), where you can review attacks, violations, and triggered signatures.
```

Reviewer 2: done

> The flow uses the NGINX One Console UI end to end. You deploy the [`secops_dashboard` log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}}) to the instance, add the F5 WAF for NGINX directives to the NGINX configuration through the console's config editor, and verify the pipeline by triggering test violations. NGINX Agent automatically configures its OpenTelemetry collector to forward security events to NGINX One Console once it sees the correct directives in the NGINX configuration — you do not need to edit the agent configuration by hand.

Reviewer 3: ```suggestion
You deploy the [`secops_dashboard` log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}}) to the instance through NGINX One Console. You then add the F5 WAF for NGINX directives to the NGINX configuration using the console's config editor, and verify the pipeline by triggering test violations. NGINX Agent automatically configures its OpenTelemetry collector to forward security events to NGINX One Console when it detects the correct directives in the NGINX configuration. You do not need to edit the NGINX Agent configuration by hand.
```

Reviewer 2: accepted

- **NGINX Agent 3.9.0 or later**: Security event forwarding to NGINX One Console requires NGINX Agent 3.9.0 or later. See the [NGINX Agent install and upgrade guide]({{< ref "/nginx-one-console/agent/install-upgrade/_index.md" >}}) to upgrade an existing agent.

---

## Deploy the `secops_dashboard` log profile

The security dashboard relies on the `secops_dashboard` log profile to capture security violations in a standardized format. It is created and maintained by F5, immutable, and pre-compiled for every available WAF compiler version, so it deploys exactly the same way as a custom log profile but completes immediately without an on-demand compile. For background, see [secops_dashboard log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}}).

1. In NGINX One Console, go to **WAF** > **Log Profiles** and select **`secops_dashboard`**.

2. From **Actions**, select **Deploy**. The **Deploy Log Profile** wizard opens.

3. In the wizard:

    - Confirm `secops_dashboard` is selected under **Log Profile**.
    - Under **Target**, choose **Instance** or **Config Sync Group** and select your target.

>     - In **Log Profile File Path**, specify the path where the compiled bundle should be deployed on the data plane (for example, `/etc/nginx/secops_dashboard.tgz`).

Reviewer 3: ```suggestion
    - In **Log Profile File Path**, specify the path where NGINX One Console should deploy the compiled bundle on the data plane (for example, `/etc/nginx/secops_dashboard.tgz`).
```

Reviewer 2: taken

> 4. Select **Next**. The wizard displays the F5 WAF for NGINX directive snippet to paste into your NGINX configuration, along with the config editor for the target instance.

Reviewer 3: ```suggestion
4. Select **Next**. The wizard displays the F5 WAF for NGINX directive snippet to paste into your NGINX configuration. The wizard also opens the config editor for the target instance.
```

Reviewer 2: done

5. Open the server block where you want to enable F5 WAF for NGINX (for example, `/etc/nginx/conf.d/default.conf`) and paste the snippet into the `server`, `http`, or `location` context. The snippet looks like this:

    ```nginx
    app_protect_security_log_enable on;
    app_protect_security_log /etc/nginx/secops_dashboard.tgz syslog:server=127.0.0.1:1514;
    ```

>     Make sure `app_protect_enable on;` and `app_protect_policy_file` are already present in the same context (covered by the prerequisites above).

Reviewer 3: ```suggestion
    `app_protect_enable on;` and `app_protect_policy_file` must be present in the same context. These are covered in [Before you begin](#before-you-begin).
```

Reviewer 2: donee

> 6. Review the configuration diff the console shows for the affected files, then select **Publish**. NGINX One Console pushes the updated configuration to the instance and reloads NGINX.

Reviewer 3: ```suggestion
6. Review the configuration for the affected files shown in the console, then select **Publish**. NGINX One Console pushes the updated configuration to the instance and reloads NGINX.
```

Reviewer 2: No the console actually shows the diff to help with review of the full configuration. Keeping the original.

For more on the deployment wizard and the alternative **Add File** > **Existing Log Profile** flow, see [Deploy log profiles]({{< ref "/nginx-one-console/waf-integration/log-profiles/deploy-log-profiles.md" >}}).

---

## Verify the setup

> When you select **Publish** in the previous step, NGINX One Console pushes the configuration change to the instance and displays a confirmation toast indicating the publish succeeded. At that point both the F5 WAF for NGINX policy and the `secops_dashboard` log profile are in place on the data plane, and the security log directive is wired up to NGINX Agent.

Reviewer 3: ```suggestion
When you select **Publish** in the previous step, NGINX One Console pushes the configuration change to the instance. NGINX One Console displays a confirmation toast indicating the publish succeeded. At that point, the F5 WAF for NGINX policy and the `secops_dashboard` log profile are in place on the data plane, and the security log directive is wired to NGINX Agent.
```

Reviewer 2: taken.

> From this point on, any request that F5 WAF for NGINX inspects on the instance produces a security event that flows to NGINX One Console. To see those events:

Reviewer 3: ```suggestion
Any request that F5 WAF for NGINX inspects on the instance produces a security event that flows to NGINX One Console. To see security events in the dashboard:
```

Reviewer 2: taken.

5. Apply a global filter on the dashboard to scope to your instance hostname or policy, in case events are present but hidden by an existing filter.

If events still do not appear after a request is processed, contact F5 support with the instance hostname and the time window you tested.

---

## References

For more information, see:

- [Security monitoring overview]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/security-monitoring-overview.md" >}})
- [secops_dashboard log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}})
- [Deploy log profiles]({{< ref "/nginx-one-console/waf-integration/log-profiles/deploy-log-profiles.md" >}})
- [Add an instance]({{< ref "/nginx-one-console/connect-instances/add-instance.md" >}})
- [Dashboard metrics reference]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/dashboard-metrics-reference.md" >}})
- [Find a security event by Support ID]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/find-event-by-support-id.md" >}})

> - [Query security events through the API]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/query-events-api.md" >}})

Reviewer 3: ```suggestion
## References

**Conceptual background**

- [Security monitoring overview]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/security-monitoring-overview.md" >}})

**Reference**

- [secops_dashboard log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}})
- [Dashboard metrics reference]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/dashboard-metrics-reference.md" >}})

**Related how-to guides**

- [Deploy log profiles]({{< ref "/nginx-one-console/waf-integration/log-profiles/deploy-log-profiles.md" >}})
- [Add an instance]({{< ref "/nginx-one-console/connect-instances/add-instance.md" >}})
- [Find a security event by Support ID]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/find-event-by-support-id.md" >}})
- [Query security events through the API]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/query-events-api.md" >}})
```

Reviewer 2: accepted
